cpp/src/replace/replace.cu

```diff
@@ -13,97 +31,840 @@
+                                                              *device_indices);
+    cudaStreamSynchronize(stream);
+
+    // Compute the offsets from the sizes
+    int32_t * offsets_data = offsets_view.data<int32_t>();
+    thrust::exclusive_scan(rmm::exec_policy(stream)->on(stream),
+                           offsets_data,
+                           offsets_data + offsets_view.size(),
+                           offsets_data);
+    int32_t size;
+    CUDA_TRY(cudaMemcpy(&size, offsets_data + offsets_view.size() - 1, sizeof(int32_t), cudaMemcpyDefault));
+    std::cout << "Allocating " << size << " bytes for new strings array\n";
+
+    // Allocate chars array and output null mask
+    std::unique_ptr<cudf::column> output_chars = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT8), size);
+    int valid_chars = cudf::bitmask_allocation_size_bytes(input_col.size());
+    rmm::device_buffer valid_bits(valid_chars, stream, mr);
+    auto output_chars_view = output_chars->mutable_view();
+    auto device_chars = cudf::mutable_column_device_view::create(output_chars_view);
+
+    replace_second<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(*device_in,
+                                                               *device_replacement,
+                                                               *device_offsets,
+                                                               *device_chars,
+                                                               *device_indices,
+                                                               reinterpret_cast<cudf::bitmask_type*>(valid_bits.data()),
+                                                               valid_count);
+    cudaStreamSynchronize(stream);
+
+    std::unique_ptr<cudf::column> output = cudf::make_strings_column(input_col.size(),
+                                                                     std::move(offsets),
+                                                                     std::move(output_chars),
+                                                                     valid_counter.value(),
+                                                                     std::move(valid_bits),
+                                                                     stream,
+                                                                     mr);
+    return output;
+  };
+};
+} //end anonymous namespace
+
+namespace cudf {
+namespace detail {
+std::unique_ptr<cudf::column> find_and_replace_all(cudf::column_view const& input_col,
+                                                   cudf::column_view const& values_to_replace,
+                                                   cudf::column_view const& replacement_values,
+                                                   rmm::mr::device_memory_resource* mr,
+                                                   cudaStream_t stream) {
+  if (0 == input_col.size() || 0 == values_to_replace.size() || 0 == replacement_values.size()) {
```

Comment:
It would be better if this condition is checked later once all `CUDF_EXPECTS` passes.

Reply:
Done.